weblink/_internal/Server.hx

```diff
@@ -45,28 +72,99 @@ class Server extends SocketServer {
 					request.pos += length;
 
 					if (request.pos >= request.length) {
-						complete(request, socket);
+						complete(request, cast client);
 						request = null;
+						Gc.enable(false);
 						return;
 					}
 				}
 
 				if (request.chunked) {
 					request.chunk(data.toString());
 					if (request.chunkSize == 0) {
-						complete(request, socket);
+						complete(request, cast client);
 						request = null;
+						Gc.enable(false);
 						return;
 					}
 				}
 
 				if (request.method != Post && request.method != Put) {
-					complete(request, socket);
+					complete(request, cast client);
 					request = null;
 				}
+
+				Gc.enable(false);
 			});
+
+			// Hashlink libuv bindings only allow for filesystem and TCP connection events.
+			// We use the fact that a new connection is opened to trigger Haxe's event loop.
+			// We have to run it on the same thread
+			// in case some of the events call (non-thread safe) libuv APIs.
+			final currentThread = Thread.current();
+			final events = currentThread.events;
+			final _ = events.progress();
+
+			Gc.enable(false); // Forbid from stopping the world when we're in uv_run
 		});
-		this.parent = parent;
+
+		// Create a thread to run the server's event loop
+		final serverThread = this.serverThread = Thread.create(() -> {
+			final currentThread = Thread.current();
+			#if (haxe_ver >= 4.3) currentThread.setName("TCP listener"); #end
+
+			// If we simply called Thread.createWithEventLoop up here,
+			// the thread would not stop after this block,
+			// but would continue running through the registered events.
+			// This way, setting Haxe's loop manually,
+			// our thread is guaranteed to eventually terminate.
+			Reflect.setProperty(currentThread, "events", new EventLoop());
+
+			this.running = true;
+			if (model == BlockUntilReady) {
+				lock.release();
+			}
+
+			Gc.enable(false);
+
+			try {
+				this.uvLoop.run(Default);
+			} catch (e:Exception) {
+				trace(e.details());
+			}
+
+			Gc.enable(true);
+
+			if (model == BlockUntilClosed) {
+				lock.release();
+			}
+		});
+
+		// Create thread #2 which will periodically wake up thread #1 with TCP connections.
+		// When the server gets no traffic, this has two side effects:
+		// 1) the Haxe event loop is still run,
+		// 2) the GC can still collect garbage.
+		if (port != 0) {
+			// Of course, this trick only works if we know the port:
+			// unfortunately, we cannot get it from a running server
+			Thread.createWithEventLoop(() -> {
+				#if (haxe_ver >= 4.3) Thread.current().setName("Timer sch. hack"); #end
+				final host = new Host("127.0.0.1");
+				final timer = this.helperTimer = new Timer(557);
+				timer.run = () -> {
+					final socket = new sys.net.Socket();
+					final _ = @:privateAccess sys.net.Socket.socket_connect(socket.__s, host.ip, port);
+					socket.close(); // Immediately close not to eat up too much resources
```

Comment:
I would feel better if this trick somehow didn't create phantom client accept connections followed by instant closes, would it be possible so that the dev facing api never sees that these connections are even taking place?

Reply:
I'm afraid I don't quite understand. `Weblink.get()`, `.post()` etc. do not see these phantom connections, because if they do not send any data, there is nothing to parse a `Request` from. Could you explain to me what you meant exactly by "dev facing api"? Thanks.

Reply:
I didn't realize this, I thought weblink might expose some type of user facing accept method, if not then I'm mistaken and this is resolved.